subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/EncryptionService.kt

```diff
@@ -179,59 +191,54 @@ class KeyStoreKeySource(
     private
     val keyProtection = KeyStore.PasswordProtection(CharArray(0))
 
+    private
+    val keyStore by lazy {
```

Comment:
note keystore is not supposed to be thread-safe. Are we guaranteeing somehow that at most one thread ever interacts with a keystore instance?

Reply:
`KeyStoreKeySource` instances are not shared between threads, the produced `SecretKey` instance might be shared but (without checking) I wouldn't expect that to be a problem.